### ISSUE TITLE
Configuring datadog-crds breaks operator.

### ISSUE THREAD

**Issue by dlorent** [DataDog/datadog-operator#721]
**Output of the info page (if this is a bug)**
```
{"level":"INFO","ts":"2023-02-27T10:48:31Z","logger":"klog","msg":"Waited for 1.04111541s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/accesscontextmanager.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:48:34Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:48:34Z","logger":"controller.datadogmonitor","msg":"Starting workers","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","worker count":1}
{"level":"INFO","ts":"2023-02-27T10:48:45Z","logger":"klog","msg":"Waited for 1.043863763s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/storage.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:48:48Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:48:55Z","logger":"klog","msg":"Waited for 1.092539577s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/events.k8s.io/v1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:48:58Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:05Z","logger":"klog","msg":"Waited for 1.141647075s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/networkservices.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:08Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:15Z","logger":"klog","msg":"Waited for 1.189249484s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/dataproc.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:18Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:25Z","logger":"klog","msg":"Waited for 1.242116826s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/flowcontrol.apiserver.k8s.io/v1beta2?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:28Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:35Z","logger":"klog","msg":"Waited for 1.293783718s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/iap.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:38Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:45Z","logger":"klog","msg":"Waited for 1.293048587s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/networking.gke.io/v1beta2?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:48Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:49:55Z","logger":"klog","msg":"Waited for 1.344080352s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/storage.k8s.io/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:49:58Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:50:05Z","logger":"klog","msg":"Waited for 1.39312779s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/serviceusage.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:50:08Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:50:15Z","logger":"klog","msg":"Waited for 1.443191219s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/pkg.crossplane.io/v1alpha1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:50:18Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"INFO","ts":"2023-02-27T10:50:26Z","logger":"klog","msg":"Waited for 1.490112356s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/accesscontextmanager.cnrm.cloud.google.com/v1beta1?timeout=32s\n"}
{"level":"ERROR","ts":"2023-02-27T10:50:28Z","logger":"controller-runtime.source","msg":"if kind is a CRD, it should be installed before calling Start","kind":"DatadogAgent.datadoghq.com","error":"no matches for kind \"DatadogAgent\" in version \"datadoghq.com/v2alpha1\""}
{"level":"ERROR","ts":"2023-02-27T10:50:30Z","logger":"controller.datadogagent","msg":"Could not wait for Cache to sync","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","error":"failed to wait for datadogagent caches to sync: timed out waiting for cache to be synced"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","msg":"Stopping and waiting for non leader election runnables"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","msg":"Stopping and waiting for leader election runnables"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"controller.datadogmonitor","msg":"Shutdown signal received, waiting for all workers to finish","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"controller.datadogmonitor","msg":"All workers finished","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","msg":"Stopping and waiting for caches"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","msg":"Stopping and waiting for webhooks"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","msg":"Wait completed, proceeding to shutdown the manager"}
{"level":"ERROR","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Problem running manager","error":"failed to wait for datadogagent caches to sync: timed out waiting for cache to be synced"}
```

**Describe what happened:**
Tried setting agent & metrics to false
  datadog-crds:
    crds:
      datadogAgents: false
      datadogMetrics: false
      datadogMonitors: true
**Describe what you expected:**
Normal running datadog-operator
(this is the output i get when setting above values to : true

```
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Version: v1.0.0-rc.8"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Build time: 2023-02-02/16:57:12"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Git Commit: 8a6fce4bc3b5cb685efa3a7c20aa584965a06b33"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Go Version: go1.18.10"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Go OS/Arch: linux/amd64"}
{"level":"INFO","ts":"2023-02-27T10:50:30Z","logger":"setup","msg":"Manager will be watching namespace","namespace":"namespace-X"}
{"level":"INFO","ts":"2023-02-27T10:50:31Z","logger":"klog","msg":"Waited for 1.038257596s due to client-side throttling, not priority and fairness, request: GET:https://10.196.0.1:443/apis/hub.gke.io/v1?timeout=32s\n"}
{"level":"INFO","ts":"2023-02-27T10:50:35Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8383"}
{"level":"INFO","ts":"2023-02-27T10:50:39Z","logger":"setup","msg":"starting manager"}
{"level":"INFO","ts":"2023-02-27T10:50:39Z","msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8383"}
{"level":"INFO","ts":"2023-02-27T10:50:39Z","msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"INFO","ts":"2023-02-27T10:50:39Z","logger":"klog","msg":"attempting to acquire leader lease infrastructure/datadog-operator-lock...\n"}


{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"klog","msg":"successfully acquired lease infrastructure/datadog-operator-lock\n"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v2alpha1.DatadogAgent"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Secret"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ConfigMap"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.DaemonSet"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Deployment"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.Role"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.RoleBinding"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ServiceAccount"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.PodDisruptionBudget"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.NetworkPolicy"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ClusterRole"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","source":"kind source: *v1.ClusterRoleBinding"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting Controller","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogmonitor","msg":"Starting EventSource","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","source":"kind source: *v1alpha1.DatadogMonitor"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogmonitor","msg":"Starting Controller","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor"}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogmonitor","msg":"Starting workers","reconciler group":"datadoghq.com","reconciler kind":"DatadogMonitor","worker count":1}
{"level":"INFO","ts":"2023-02-27T10:51:44Z","logger":"controller.datadogagent","msg":"Starting workers","reconciler group":"datadoghq.com","reconciler kind":"DatadogAgent","worker count":1}
```

**Steps to reproduce the issue:**
configure the values in helm to disable datadogAgents, datadogMetrics

**Additional environment details (Operating System, Cloud provider, etc):**
GKE: v1.24.9-gke.2000
Datadog-Operator: Version: v1.0.0-rc.8

**Comment by CharlyF** [comment]
Hi @dlorent - Thanks for raising this. 
I will look into it now.

**Comment by CharlyF** [comment]
Hi @dlorent  - I wanted to circle back on this. We released the Datadog Operator 1.0 yesterday and updated the `datadog-crds` chart to 1.0.0 as well as the `datadog-operator` chart to 1.0.0.
This change should be available now, please let us know how it goes.

NB: Please let us know if you have feedback about the migration process, we tried to document it in the public doc, in the helm repo and in this repo but I want to make sure it goes smoothly for all our users.

**Comment by dlorent** [comment]
> Hi @dlorent - I wanted to circle back on this. We released the Datadog Operator 1.0 yesterday and updated the `datadog-crds` chart to 1.0.0 as well as the `datadog-operator` chart to 1.0.0. This change should be available now, please let us know how it goes.
> 
> NB: Please let us know if you have feedback about the migration process, we tried to document it in the public doc, in the helm repo and in this repo but I want to make sure it goes smoothly for all our users.

Hi @CharlyF 
Thank you for the effort :) I will test it within the next few days, and let you know.

**Comment by CharlyF** [comment]
Great to hear! I will close this for now but feel free to reach out if you have questions!
